---
title: "Class vs T, Sal, NO3, distributions"
author: "Andres"
date: "17/05/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:


#Libraries

```{r cars}
  library(stringr)
   library(ggplot2)
   library(dplyr)
   library(tidyr)
   library(tibble)
   library(readr)
   library(maps)
   library(glue)
   library("cowplot")
   library("treemapify")
library("Cairo")
library("cairoDevice")
   library(phyloseq)
library("readxl")
library("openxlsx")
library("gplots")
library(forcats)
library(mgcv)
library(gplots)



# install.packages("tidyverse")
# 
# install.packages("remotes")
# remotes::install_github("hadley/dplyr")
# 
#    # library(pr2database)
#    # data("pr2")
```

#Load the phyloseq files

```{r pressure, echo=FALSE}
  
  #ps <- list()
  #ps[["all"]] <- readRDS("intomob123_phyloseq_2April2020_RANKED.rds") 
  
  ps <- readRDS("intomob123_phyloseq_15March2021_RANKED.rds")
```
  

# Retain seawater samples, and those from voyages for which we have complete metadata

```{r pressure, echo=FALSE}
ps_sw <-  ps %>%  
  subset_samples(Sample_Type == "Seawater") %>%
  #subset_samples((Project != "SOAP" )) %>%
  #subset_samples((Project != "Bay of Plenty" )) %>%
  ### Remove SOAP samples until I get metadata complete
      subset_samples(Water_mass.TS1 != "ANT" ) %>%
  subset_samples((Sampling_Type != "Blank" )) ### Remove SOAP samples until I get metadata complete
```

## ELiminate taxa with zero reads
```{r cars} 
 ps_sw <-  ps_sw %>%  
     filter_taxa(function(x) sum(x) > 0 , TRUE) 
 ps_sw
```

# Filtration based on Taxonomy - 


# for PROTISTS

```{r pressure, echo=FALSE}
  # Remove Metazoa
  # Remove Fungi

ps_protist <-  ps_sw %>%  
  subset_taxa(!(Division %in% c("Metazoa", "Fungi"))) 
```

# for PROTIST -SYNDINALES
```{r pressure, echo=FALSE}
  # Remove Metazoa
  # Remove Fungi
  # Protist - syndiniales and sarcomonadae
  
ps_protist_synd <-  ps_sw %>%  
        subset_taxa(!(Division %in% c("Metazoa", "Fungi"))) %>%
      subset_taxa(!(Class %in% c("Syndiniales", "Sarcomonadea")))

get_taxa_unique(ps_protist_synd, "Class")

```

# Colors for plotting taxonomy

```{r pressure, echo=FALSE}

classes_division <- read.table("Color_division.csv", sep=",", header=T, check.names = F, comment.char = "")

classes_division$color_hex_division <-col2hex(classes_division$color_division)
class_color_division <- structure(as.character(classes_division$color_division),.Names=as.character(classes_division$division))

classes_division$color_hex_class <-col2hex(classes_division$color_class)
class_color_class <- structure(as.character(classes_division$color_hex_class),.Names=as.character(classes_division$class))
```


# Normalize the otu table - NORMALIZED READ ABUNDANCES
# Define function - phyloseq_normalize_median

```{r pressure, echo=FALSE}

## Standardize abundances to the median sequencing depth

#PROTIST
ps_protist_median = median(sample_sums(ps_protist))
normalize_protist_median = function(x, t=ps_protist_median) round(t * (x / sum(x)))
ps_protist_std = transform_sample_counts(ps_protist, normalize_protist_median)

#PROTIST-SYND
ps_protist_synd_median = median(sample_sums(ps_protist_synd))
normalize_protist_synd_median = function(x, t=ps_protist_synd_median) round(t * (x / sum(x)))
ps_protist_synd_std = transform_sample_counts(ps_protist_synd, normalize_protist_synd_median)


```

## Agglomerate reads into CLASSES only 30 more normalized abundant classes

```{r pressure, echo=FALSE}

# DIVISION
ps_protist_division <- tax_glom(ps_protist_std, taxrank = "Division")
ps_protist_synd_division <- tax_glom(ps_protist_synd_std, taxrank = "Division")

# CLASS
ps_protist_class <- tax_glom(ps_protist_std, taxrank = "Class")
ps_protist_synd_class <- tax_glom(ps_protist_synd_std, taxrank = "Class")

plot(sort(taxa_sums(ps_protist_synd_class), TRUE) )
get_taxa_unique(ps_protist, "Class")
get_taxa_unique(ps_protist_synd_class, "Class")

### Top30 - most abundant classes
 class.sum = tapply(taxa_sums(ps_protist_synd_class), tax_table(ps_protist_synd_class)[, "Class"], sum, na.rm=TRUE)
  top30class = names(sort(class.sum, TRUE))[1:30]
  Top30class = prune_taxa((tax_table(ps_protist_synd_class)[, "Class"] %in% top30class), ps_protist_synd_class)
  Top30class
  
Class_names = data.frame(Class_reads = sort(taxa_sums(Top30class), TRUE), sorted = 1:ntaxa(Top30class), Class = tax_table(Top30class)[, "Class" ])

write.xlsx(Class_names , file = "Tables/Top30_Class_read_abundance_sw.xlsx")

saveRDS(ps_protist_synd_class,str_c("intomob123_phyloseq_15March2021_RANKED.rds_RANKED_CLASS.rds"))

   
ps_protist_synd_class <- readRDS("intomob123_phyloseq_15March2021_RANKED.rds_RANKED_CLASS.rds") ### load previously saved phyloseq object

```
## ANALYSIS OF RELATIVE ABUNDANCES - %
##Calculate statistics of group specific relative abundances 

# Agglomerate - 
# Normalize the otu table 
# Define function - phyloseq_normalize_median

```{r pressure, echo=FALSE}

# First - agglomerate the sequencing depth normalized OTU table 
          # ALREADY DONE above 

## Second - Standardize to relative abundances 

#PROTIST - CLASS and DIVISION
ps_protist_class.rel = transform_sample_counts(ps_protist_class, function(OTU) OTU*100/sum(OTU))

ps_protist_division.rel = transform_sample_counts(ps_protist_division, function(OTU) OTU*100/sum(OTU))

#PROTIST-SYND - CLASS and DIVISION
ps_protist_synd_class.rel = transform_sample_counts(ps_protist_synd_class, function(OTU) OTU*100/sum(OTU))

ps_protist_synd_division.rel = transform_sample_counts(ps_protist_synd_division, function(OTU) OTU*100/sum(OTU))

## Select TOP20 classes

### OPTION A - select PROTIST classes of interest
ps_protist_synd_class.rel.1 <- ps_protist_synd_class.rel %>%
   subset_taxa(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" ))
 
### OPTION B - select PROTIST-SYND classes of interest based on based on RELATIVE abundance
protist.class.sum = tapply(taxa_sums(ps_protist_synd_class.rel), tax_table(ps_protist_synd_class.rel)[, "Class"], sum, na.rm=TRUE)

top20class = names(sort(protist.class.sum, TRUE))[1:20]

ps_protist_class.top20 = prune_taxa((tax_table(ps_protist_synd_class.rel)[, "Class"] %in% top20class), ps_protist_synd_class.rel)
  ps_protist_class.top20
  
Class_names1 = data.frame(Class_reads = sort(taxa_sums(ps_protist_class.top20), TRUE), sorted = 1:ntaxa(ps_protist_class.top20), Class = tax_table(ps_protist_class.top20)[, "Class" ])

```

## Make LONG FORMAT dataframe for plotting

## PROTIST-SYND

```{r pressure, echo=FALSE}

## dataframe for plotting
otu_df <- as.data.frame(ps_protist_class.top20@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      #filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

taxo_df <- as.data.frame(ps_protist_class.top20@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

otu_df <- left_join(taxo_df, otu_df)
metadata_df <- data.frame(sample_data(ps_protist_class.top20)) %>%
      rownames_to_column(var = "file_code")
otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))

df_protist_synd_class.rel <-left_join(otu_df, classes_division, by=c("Class"= "class"))
df_protist_synd_class.rel <- df_protist_synd_class.rel %>%
      mutate(pico_pct = Chla_02*100/(Chla_02 + Chla_2 + Chla_20)) %>%
      mutate(nano_pct = Chla_2*100/(Chla_02 + Chla_2 + Chla_20)) %>%
      mutate(micro_pct = Chla_20*100/(Chla_02 + Chla_2 + Chla_20))
    
    # write.xlsx(df_protist_synd_class.rel, file = "Tables/df_protist_synd_class.rel.xlsx")

```

## Summary mean and sd relative abundance of classes in different TOTAL and in WATER MASSES 

### TOTAL DATASET EEZ 
```{r, fig.height=4, fig.width=6}

# FILTER ONLY EUPHOTIC
class_df_protist_stat_total <-  df_protist_synd_class.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Class) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

 class_df_protist_stat_total <- class_df_protist_stat_total %>%
  arrange( -mean) 

write.xlsx(class_df_protist_stat_total, file = "Tables/Protist_synd_euphotic_top20_CLASS_REL_STATS_CLASSES_March2021.xlsx")

## BAR PLOT - Figure 5A
# color coding # Water mass - SUM
class_df_protist_stat_total_col <-left_join(class_df_protist_stat_total, classes_division, by=c("Class"= "class")) # join for colors

color_class <- as.character(class_df_protist_stat_total_col$color_hex_class)
names(color_class) <- as.character(class_df_protist_stat_total_col$Class)
class_df_protist_stat_total_col$Class = with(class_df_protist_stat_total_col, reorder(Class, -mean))

CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_CLASS_REL_STATS_CLASSES_March2021_Fig_5A.pdf", width = 8, height = 6)
  ggplot(class_df_protist_stat_total_col, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - MEAN")) + 
    geom_errorbar(aes(ymin = mean, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.1, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.1)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(0, 60)
dev.off()


```
## WATER MASSES - Average +/- SD of Top20 classes
```{r, fig.height=4, fig.width=6}

# WATER MASSES - EUPHOTIC
class_df_protist_stat_watermass <-  df_protist_synd_class.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Water_mass.TS1, Class) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

 class_df_protist_stat_watermass <- class_df_protist_stat_watermass %>%
  arrange(Water_mass.TS1, -mean) 

write.xlsx(class_df_protist_stat_watermass, file = "Tables/class_df_protist_stat_REL_WATERMASSES_March2021.xlsx")


```

#### BAR PLOT - ALL WATER MASSES PLOTTED TOGETHER -  Figure 5A and 5B
```{r, fig.height=4, fig.width=6}

# color coding # Water mass 
class_df_protist_stat_watermass <-left_join(class_df_protist_stat_watermass, classes_division, by=c("Class"= "class")) # join for colors


CairoPDF(file = "R.Figures/Bar/class_df_protist_stat_top20_REL_WATERMASSES_March2021.pdf", width = 6, height = 9)
ggplot(class_df_protist_stat_watermass, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - SUM")) + 
    geom_errorbar(aes(ymin = mean, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(0, 60) +
    facet_wrap(~Water_mass.TS1, nrow = 3)

dev.off()

```

## PLOT DIFFERENT WATER MASSES INDIVIDUALLY - Figure 5A and 5B
```{r, fig.height=4, fig.width=6}

## STW ##
CairoPDF(file = "R.Figures/Bar/class_df_protist_stat_top20_REL_STW.pdf", width = 8, height = 6)

class_df_protist_stat_stw <- class_df_protist_stat_watermass %>%
  filter(Water_mass.TS1 == "STW") %>%
  arrange(-mean)

color_class_stw <- as.character(class_df_protist_stat_stw$color_hex_class)
names(color_class_stw) <- as.character(class_df_protist_stat_stw$Class)
class_df_protist_stat_stw$Class = with(class_df_protist_stat_stw, reorder(Class, -mean))
  
    ggplot(class_df_protist_stat_stw, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class_stw) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - STW")) + 
    geom_errorbar(aes(ymin = mean, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(0, 60) 
dev.off()

## SAW ##
CairoPDF(file = "R.Figures/Bar/class_df_protist_stat_top20_REL_SAW.pdf", width = 8, height = 6)

class_df_protist_stat_saw <- class_df_protist_stat_watermass %>%
  filter(Water_mass.TS1 == "SAW") %>%
  arrange(-mean)

color_class_saw <- as.character(class_df_protist_stat_saw$color_hex_class)
names(color_class_saw) <- as.character(class_df_protist_stat_saw$Class)
class_df_protist_stat_saw$Class = with(class_df_protist_stat_saw, reorder(Class, -mean))
  
    ggplot(class_df_protist_stat_saw, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class_saw) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - SAW")) + 
    geom_errorbar(aes(ymin = mean, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(0, 60) 
dev.off()

## STF ##
CairoPDF(file = "R.Figures/Bar/class_df_protist_stat_top20_REL_STF.pdf", width = 8, height = 6)

class_df_protist_stat_stf <- class_df_protist_stat_watermass %>%
  filter(Water_mass.TS1 == "STF") %>%
  arrange(-mean)

color_class_stf <- as.character(class_df_protist_stat_stf$color_hex_class)
names(color_class_stf) <- as.character(class_df_protist_stat_stf$Class)
class_df_protist_stat_stf$Class = with(class_df_protist_stat_stf, reorder(Class, -mean))
  
    ggplot(class_df_protist_stat_stf, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class_stf) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - STF")) + 
    geom_errorbar(aes(ymin = mean, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(0, 60) 

dev.off()

```

## DIVISION ### - Figure 5A and 5B

## PROTIST - SYND
```{r pressure, echo=FALSE}

### OPTION A - select PROTIST-SYND classes of interest based on based on RELATIVE ABUNDANCE (same 20 groups as plotted above)

protist.division.sum = tapply(taxa_sums(ps_protist_synd_division.rel), tax_table(ps_protist_synd_division.rel)[, "Division"], sum, na.rm=TRUE)
  top10class = names(sort(protist.division.sum, TRUE))[1:15]
  ps_protist_division = prune_taxa((tax_table(ps_protist_synd_division.rel)[, "Division"] %in% top10class), ps_protist_synd_division.rel)
  ps_protist_division
  
  ## PROTIST - SYND

## dataframe for plotting
otu_df <- as.data.frame(ps_protist_division@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      #filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

taxo_df <- as.data.frame(ps_protist_division@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

otu_df <- left_join(taxo_df, otu_df)
metadata_df <- data.frame(sample_data(ps_protist_division)) %>%
      rownames_to_column(var = "file_code")
otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))
df_protist_division.rel <-left_join(otu_df, classes_division, by=c("Division"= "division"))
    
    # write.xlsx(df_protist_division.rel, file = "Tables/df_protist_synd_division.rel_March2021.xlsx")
    #   
```

## Summary mean and sd relative abundance of classes in different water masses
### DIVISION - TOTAL - EEZ #####
```{r, fig.height=6, fig.width=6}

## PRotist-Synd DIVISION
division_df_protist_stat_total <-  df_protist_division.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Division) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

 division_df_protist_stat_total <- division_df_protist_stat_total %>%
  arrange(-mean) 

write.xlsx(division_df_protist_stat_total, file = "Tables/Protist_synd_euphotic_top20_CLASS_REL_STATS_DIVISION_March2021.xlsx")

## BAR PLOT
# color coding # Water mass - SUM
division_df_protist_stat_total_col <-left_join(division_df_protist_stat_total, classes_division, by=c("Division"= "division")) %>%
distinct(division_df_protist_stat_total_col, Division, .keep_all = TRUE)

color_division <- as.character(division_df_protist_stat_total_col$color_hex_division)
names(color_division) <- as.character(division_df_protist_stat_total_col$division)
division_df_protist_stat_total_col$Division = with(division_df_protist_stat_total_col, reorder(Division, -mean))


CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_REL_DIVISION.pdf", width = 7, height = 6.5)
ggplot(division_df_protist_stat_total_col, aes(x=Division, y=mean)) +
    geom_bar(stat="identity", width = 0.9) +
   # scale_fill_manual(values = color_division) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - REL")) + 
    geom_errorbar(aes(ymin = mean, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(0, 60)
dev.off()

```


## Summary mean and sd relative abundance of classes in different water masses
### WATER MASSES  #####
```{r, fig.height=6, fig.width=6}

## PRotist DIVISION
division_df_protist_stat_watermass <-  df_protist_division.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Water_mass.TS1, Division) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

division_df_protist_stat_watermass <- division_df_protist_stat_watermass %>%
  arrange(Water_mass.TS1, -mean) 

write.xlsx(division_df_protist_stat_watermass, file = "Tables/division_df_protist_stat_REL_WATERMASSES_March2021.xlsx")


CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_REL_DIVISION_WATERMASS2.pdf", width = 4, height = 9)
ggplot(division_df_protist_stat_watermass_col, aes(x=Division, y=mean)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_division) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - MEAN")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60) +
    facet_wrap(~Water_mass.TS1, nrow = 3)

dev.off()

```


### CLASS LEVEL ABUNDANCE/DISTRIBUTION ANALYSIS #########

# Select main classes of interest  -  most abundant classes
```{r pressure, echo=FALSE}

### OPTION A - select PROTIST classes of interest
ps_protist_class.1 <- ps_protist_synd_class %>%
   subset_taxa(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" ))
 

### OPTION B - select PROTIST-SYND classes of interest based on based on RELATIVE ABUNDANCE (same 20 groups as plotted above) 

## Already selected groups above - use the classes selected based on REL ABUND

protist.class.sum = tapply(taxa_sums(ps_protist_synd_class.rel), tax_table(ps_protist_synd_class.rel)[, "Class"], sum, na.rm=TRUE)
top20class = names(sort(protist.class.sum, TRUE))[1:20]

ps_protist_class.top20.abund = prune_taxa((tax_table(ps_protist_synd_class)[, "Class"] %in% top20class), ps_protist_synd_class)
  ps_protist_class.top20.abund
  
Class_names.abund = data.frame(Class_reads = sort(taxa_sums(ps_protist_class.top20.abund), TRUE), sorted = 1:ntaxa(ps_protist_class.top20.abund), Class = tax_table(ps_protist_class.top20.abund)[, "Class" ])

```
## Make LONG FORMAT dataframe for plotting

```{r pressure, echo=FALSE}

## dataframe for plotting -- Look into standardizes or relative abundances
otu_df <- as.data.frame(ps_protist_class.top20.abund@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      #filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

taxo_df <- as.data.frame(ps_protist_class.top20.abund@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

    otu_df <- left_join(taxo_df, otu_df)

    metadata_df <- data.frame(sample_data(ps_protist_class.top20.abund)) %>%
      rownames_to_column(var = "file_code")

    otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))
    
    class_df_protist_synd.abund <-left_join(otu_df, classes_division, by=c("Class"= "class"))
    class_df_protist_synd.abund <- class_df_protist_synd.abund %>%
      mutate(pico_pct = Chla_02*100/(Chla_02 + Chla_2 + Chla_20)) %>%
      mutate(nano_pct = Chla_2*100/(Chla_02 + Chla_2 + Chla_20)) %>%
      mutate(micro_pct = Chla_20*100/(Chla_02 + Chla_2 + Chla_20))
    
    # write.xlsx(class_df_protist_synd.abund, file = "Tables/class_df_protist_synd_abund.xlsx")

      
```




## Plotting BOX-PLOTS of groups relative abundance in different areas and environmental ranks 
```{r, fig.height=6, fig.width=6}


###PLOTTING - BOX PLOT of relative abundances in different areas, and other RANKED ENVIRONMENTAL VARIABLES

## WATER MASSES - Figure 6

# I would like to organize the group specific plots from more to less abundant but I don't know yet how to do this using facet-wrap 

  CairoPDF(file = "R.Figures/BoxPlot/Protist_Synd_Abundance_EUPHOTIC_Water_masses.pdf", width = 14, height = 8) 
  
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
ggplot(aes(x = Water_mass.TS1 , y = n_reads,  fill(Water_mass.TS1), color = Water_mass.TS1)) +
        #geom_point(size = 1) +
       geom_boxplot(outlier.shape = NA) + 
        #geom_boxplot() +
      geom_jitter(size = 0.8, width = 0.2, alpha = 0.2) +
        #scale_x_continuous(limits = c(0, 10)) +
        #ylim(-100, 0) +
        #scale_color_manual(values = water_colors) +
       # geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
        theme_bw() + ggtitle("Class - EUPHOTIC zone PROTIST-SYND normalized read abundance in Water.masses") +
        xlab("Water_mass.TS1")+ ylab("standardized read abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 12)) +
        theme(axis.text.y = element_text(size = 12)) +
        facet_wrap(~Class, scales = "free_y")
  dev.off()
  
    ### AREAS ## - Figure S6
## reorder areas ##
  
CairoPDF(file = "R.Figures/BoxPlot/Protist_Synd_Abundance_EUPHOTIC_Area.pdf", width = 14, height = 8)
  
  class_df_protist_synd.abund %>%
    mutate(area.name = fct_reorder(Area2, Latitude.CTD)) %>% ## creating a new variable to reorder factors
  
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  #filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
   
    ggplot(aes(x = area.name , y = n_reads,  fill(Water_mass.TS1), color = Water_mass.TS1)) +
        #geom_point(size = 1) +
       geom_boxplot(outlier.shape = NA) + 
        #geom_boxplot() +
      geom_jitter(size = 0.8, width = 0.2, alpha = 0.2) +
        #scale_x_continuous(limits = c(0, 10)) +
        #ylim(-100, 0) +
        #scale_color_manual(values = water_colors) +
       # geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
        theme_bw() + ggtitle("Class - EUPHOTIC zone standardized abundance by AREA") +
        xlab("Area")+ ylab("Standarized read abundance") +
        theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 12)) +
        theme(axis.text.y = element_text(size = 12)) +
        facet_wrap(~Class, scales = "free_y")
  dev.off()
  

```

## Plotting distribution of groups MEAN SEQUENCING DEPTH normalized abundances and environmental factors

## Using Geom_smooth - method = loess model y - x 

## As it is I don't find these very informative
## Same analysis can be plotted within different water masses to explore how different groups relate to environmental drivers within each water mass/biome -  but still not clear trends.

## Could explore how to use GAM with multiple environmental parameters (similar to Tristan Biard L&O 2020 paper with Collodaria)

### TEMPERATURE
```{r, fig.height=10, fig.width=10}

  CairoPDF(file = "R.Figures/Environmental/Class_vs_Temperature_MIXEDLAYER_watermass-ALL.pdf", width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
    ggplot(aes(x = Temperature.CTD , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5, size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs TEMPERATURE") +
        xlab("Salinity")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        #facet_wrap(~Class, scales = "free") 
        theme(strip.text = element_text(size = 12)) +
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y") +
      facet_wrap(~Class, scales = "free_y")
  dev.off()
  
  
```

### SALINITY
```{r, fig.height=10, fig.width=10}

  CairoPDF(file = "R.Figures/Environmental/Class_vs_Salinity_MIXEDLAYER_watermass-ALL.pdf", width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
    ggplot(aes(x = Salinity.CTD , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5, size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs SALINITY") +
        xlab("Salinity")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        #facet_wrap(~Class, scales = "free") 
        theme(strip.text = element_text(size = 12)) +
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y") +
      facet_wrap(~Class, scales = "free_y")
  dev.off()
  
  
```

## NITRATE ##
```{r, fig.height=10, fig.width=6}

## sample water concentration
CairoPDF(file = "R.Figures/Environmental/Class_vs_NO3_MIXEDLAYER_watermass-ALL.pdf", width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
    ggplot(aes(x = NO3 , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5, size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        xlab("Nitrate concentration (umol/L)")+ ylab("normalized abundance") +
        theme_bw() + ggtitle("EUPHOTIC - Class normalized abundance vs NO3") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
       # facet_wrap(~Class, scales = "free") +
        theme(strip.text = element_text(size = 12)) +
        # facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y")
     facet_wrap(~Class, scales = "free_y")

  dev.off()
  
  
  ## Water column median waterconcentration

  CairoPDF(file = "R.Figures/Environmental/Class_vs_medianNO3_EUPHOTIC_watermass-ALL.pdf",width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  #filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
    ggplot(aes(x = median_NO3 , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("EUPHOTIC - normalized abundance vs median NO3") +
        xlab("Nitrate concentration (umol/L)")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
       # facet_wrap(~Class, scales = "free") +
        theme(strip.text = element_text(size = 12))  +
           # facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y")
     facet_wrap(~Class, scales = "free_y")
  dev.off()
  
```
## CHLA ##
```{r, fig.height=10, fig.width=6}

  CairoPDF(file = "R.Figures/Environmental/Class_vs_Chla_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
    ggplot(aes(x = ChlaTot , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("EUPHOTIC - Class normalized abundance vs Chla") +
        xlab("Chla concentration (ug/L)")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        #facet_wrap(~Class, scales = "free") 
        # facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y")
     facet_wrap(~Class, scales = "free_y")
  dev.off()
  
```
## SF - CHLA normalized abundance  vs  chla percentage ##

```{r, fig.height=10, fig.width=6}

## PICO ##
  CairoPDF(file = "R.Figures/Environmental/Class_vs_SFChla_0.2-2_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
    ggplot(aes(x = pico_pct , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs Chla 0.2-2") +
        xlab("% Chla 0.2-2 um ")+ ylab("normalized abundance") +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        #facet_wrap(~Class, scales = "free") 
       # facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y")
     facet_wrap(~Class, scales = "free_y")
  dev.off()
  
  
```

```{r, fig.height=10, fig.width=6}

## NANO ##
  CairoPDF(file = "R.Figures/Environmental/Class_vs_SFChla_2-20_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
  
    ggplot(aes(x = nano_pct , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs Chla 2-20") +
        xlab("% Chla 2-20 um ")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        #facet_wrap(~Class, scales = "free") 
       # facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y")
     facet_wrap(~Class, scales = "free_y")
  dev.off()
  
  
```

```{r, fig.height=10, fig.width=6}

## MICRO ##
  CairoPDF(file = "R.Figures/Environmental/Class_vs_SFChla_20_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  class_df_protist_synd.abund %>%
  #filter(NominalDepth == 1) %>%
  filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  #filter(Water_mass.TS1 != "ANT" ) %>% 
  # filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%
  
  
    ggplot(aes(x = micro_pct , y = n_reads, group = Class, color = Water_mass.TS1)) +
         geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs Chla >20") +
        xlab("% Chla >20 um ")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        #facet_wrap(~Class, scales = "free") 
       # facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales = "free_y")
     facet_wrap(~Class, scales = "free_y")
  dev.off()
  
  
```

## Calculate statistics of group specific relative abundances - %

# Agglomerate - 
# Normalize the otu table 
# Define function - phyloseq_normalize_median

```{r pressure, echo=FALSE}

# First - agglomerate the sequencing depth normalized OTU table
# ALREADY DONE FOR CLASS standardized abundance ANALYSiS above -  Not for DIVIsION

ps_protist_class <- tax_glom(ps_protist_std, taxrank = "Class")
ps_phyto_class <- tax_glom(ps_phyto_std, taxrank = "Class")
ps_protist_synd_class <- tax_glom(ps_protist_synd_std, taxrank = "Class")


ps_protist_division <- tax_glom(ps_protist_std, taxrank = "Division")
ps_phyto_division <- tax_glom(ps_phyto_std, taxrank = "Division")
ps_protist_synd_division <- tax_glom(ps_protist_synd_std, taxrank = "Division")


## Second - Standardize to relative abundances 
#PHYTO
ps_phyto_class.rel = transform_sample_counts(ps_phyto_class, function(OTU) OTU*100/sum(OTU))
ps_phyto_division.rel = transform_sample_counts(ps_phyto_division, function(OTU) OTU*100/sum(OTU))

#PROTIST
ps_protist_class.rel = transform_sample_counts(ps_protist_class, function(OTU) OTU*100/sum(OTU))
ps_protist_division.rel = transform_sample_counts(ps_protist_division, function(OTU) OTU*100/sum(OTU))

#PROTIST-SYND
ps_protist_synd_class.rel = transform_sample_counts(ps_protist_synd_class, function(OTU) OTU*100/sum(OTU))
ps_protist_synd_division.rel = transform_sample_counts(ps_protist_synd_division, function(OTU) OTU*100/sum(OTU))

## Select TOP15 classes

### OPTION A1 - select PROTIST classes of interest
ps_protist_synd_class.rel.1 <- ps_protist_synd_class.rel %>%
   subset_taxa(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" ))
 
  
### OPTION C - select PROTIST-SYND classes of interest based on based on TOTAL abundanc
protist.class.sum = tapply(taxa_sums(ps_protist_synd_class.rel), tax_table(ps_protist_synd_class.rel)[, "Class"], sum, na.rm=TRUE)
  top10class = names(sort(protist.class.sum, TRUE))[1:15]
  ps_protist_class.3 = prune_taxa((tax_table(ps_protist_synd_class.rel)[, "Class"] %in% top10class), ps_protist_synd_class.rel)
  ps_protist_class.3

```
## Make LONG FORMAT dataframe for plotting

## PROTIST-SYND

```{r pressure, echo=FALSE}

## dataframe for plotting
otu_df <- as.data.frame(ps_protist_synd_class.rel.1@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      #filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

taxo_df <- as.data.frame(ps_protist_synd_class.rel.1@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

otu_df <- left_join(taxo_df, otu_df)
metadata_df <- data.frame(sample_data(ps_protist_synd_class.rel.1)) %>%
      rownames_to_column(var = "file_code")
otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))

df_protist_synd_class.rel <-left_join(otu_df, classes_division, by=c("Class"= "class"))
df_protist_synd_class.rel <- df_protist_synd_class.rel %>%
      mutate(pico_pct = Chla_02*100/(Chla_02 + Chla_2 + Chla_20)) %>%
      mutate(nano_pct = Chla_2*100/(Chla_02 + Chla_2 + Chla_20)) %>%
      mutate(micro_pct = Chla_20*100/(Chla_02 + Chla_2 + Chla_20))
    
    write.xlsx(df_protist_synd_class.rel, file = "Tables/df_protist_synd_class.rel.xlsx")

```

## Summary mean and sd relative abundance of classes in different TOTAL and in WATER MASSES 

### TOTAL DATASET EEZ 
```{r, fig.height=4, fig.width=6}

# FILTER ONLY EUPHOTIC
class_df_protist_stat_total <-  df_protist_synd_class.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Class) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

 class_df_protist_stat_total <- class_df_protist_stat_total %>%
  arrange( -mean) 

write.xlsx(class_df_protist_stat_total, file = "Tables/class_df_protist_synd_stat_TOTAL_August.xlsx")

## BAR PLOT
# color coding # Water mass - SUM
class_df_protist_stat_total_col <-left_join(class_df_protist_stat_total, classes_division, by=c("Class"= "class")) # join for colors

color_class <- as.character(class_df_protist_stat_total_col$color_hex_class)
names(color_class) <- as.character(class_df_protist_stat_total_col$Class)
class_df_protist_stat_total_col$Class = with(class_df_protist_stat_total_col, reorder(Class, -mean))


CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_SUM_CLASSES.pdf", width = 6, height = 4)
ggplot(class_df_protist_stat_total_col, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - SUM")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60)
dev.off()
 
  #ggsave("Protist_synd_euphotic_top20_SUM_CLASSES.pdf", path = "R.Figures/Bar", dpi = 300, height = 6, width = 8)


```
## WATER MASSES
```{r, fig.height=4, fig.width=6}

# WATER MASSES 
class_df_protist_stat_watermass <-  df_protist_synd_class.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Water_mass.TS1, Class) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

 class_df_protist_stat_watermass <- class_df_protist_stat_watermass %>%
  arrange(Water_mass.TS1, -mean) 

write.xlsx(class_df_protist_stat_watermass, file = "Tables/class_df_protist_stat__WATER_August.xlsx")

## BAR PLOT
# color coding # Water mass - SUM
class_df_protist_stat_watermass <-left_join(class_df_protist_stat_watermass, classes_division, by=c("Class"= "class")) # join for colors

color_class_stw <- as.character(class_df_protist_stat_watermass$color_hex_class)
names(color_class_stw) <- as.character(class_df_protist_stat_watermass$Class)
class_df_protist_stat_watermass$Class = with(class_df_protist_stat_watermass, reorder(Class, -mean))


CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_SUM__WATERMASS2.pdf", width = 6, height = 9)
ggplot(class_df_protist_stat_watermass, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - SUM")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60) +
    facet_wrap(~Water_mass.TS1, nrow = 3)

dev.off()

# INDIVIDUALLY BY WATER MASS #

## STW ##
CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_STW.pdf", width = 6, height = 4)

class_df_protist_stat_stw <- class_df_protist_stat_watermass %>%
  filter(Water_mass.TS1 == "STW") %>%
  arrange(-mean)

color_class_stw <- as.character(class_df_protist_stat_stw$color_hex_class)
names(color_class1) <- as.character(class_df_protist_stat_stw$Class)
class_df_protist_stat_stw$Class = with(class_df_protist_stat_stw, reorder(Class, -mean))
  
    ggplot(class_df_protist_stat_stw, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class_stw) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - STW")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60) 
dev.off()

## SAW ##
CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_SAW.pdf", width = 6, height = 4)

class_df_protist_stat_saw <- class_df_protist_stat_watermass %>%
  filter(Water_mass.TS1 == "SAW") %>%
  arrange(-mean)

color_class_saw <- as.character(class_df_protist_stat_saw$color_hex_class)
names(color_class_saw) <- as.character(class_df_protist_stat_saw$Class)
class_df_protist_stat_saw$Class = with(class_df_protist_stat_saw, reorder(Class, -mean))
  
    ggplot(class_df_protist_stat_saw, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class_saw) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - SAW")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60) 
dev.off()

## STF ##
CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_STF.pdf", width = 6, height = 4)

class_df_protist_stat_stf <- class_df_protist_stat_watermass %>%
  filter(Water_mass.TS1 == "STF") %>%
  arrange(-mean)

color_class_stf <- as.character(class_df_protist_stat_stf$color_hex_class)
names(color_class_stf) <- as.character(class_df_protist_stat_stf$Class)
class_df_protist_stat_stf$Class = with(class_df_protist_stat_stf, reorder(Class, -mean))
  
    ggplot(class_df_protist_stat_stf, aes(x=Class, y=mean, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class_stf) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - STF")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60) 

dev.off()


```

## DIVISION ###


## PROTIST
```{r pressure, echo=FALSE}

### OPTION C - select PROTIST-SYND classes of interest based on based on TOTAL abundanc
protist.division.sum = tapply(taxa_sums(ps_protist_synd_division.rel), tax_table(ps_protist_synd_division.rel)[, "Division"], sum, na.rm=TRUE)
  top10class = names(sort(protist.division.sum, TRUE))[1:15]
  ps_protist_division = prune_taxa((tax_table(ps_protist_synd_division.rel)[, "Division"] %in% top10class), ps_protist_synd_division.rel)
  ps_protist_division
  
  ## PROTIST

## dataframe for plotting
otu_df <- as.data.frame(ps_protist_division@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      #filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

taxo_df <- as.data.frame(ps_protist_division@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

otu_df <- left_join(taxo_df, otu_df)
metadata_df <- data.frame(sample_data(ps_protist_division)) %>%
      rownames_to_column(var = "file_code")
otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))
df_protist_division.rel <-left_join(otu_df, classes_division, by=c("Division"= "division"))
    
    write.xlsx(df_protist_division.rel, file = "Tables/df_protist_synd_division.rel_August.xlsx")
      
```

## Summary mean and sd relative abundance of classes in different water masses
### DIVISION - TOTAL - EEZ #####
```{r, fig.height=6, fig.width=6}

## PRotist DIVISION
division_df_protist_stat_total <-  df_protist_division.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Division) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

 division_df_protist_stat_total <- division_df_protist_stat_total %>%
  arrange(-mean) 

write.xlsx(division_df_protist_stat_total, file = "Tables/division_df_protist_stat_EEZ_August.xlsx")

## BAR PLOT
# color coding # Water mass - SUM
division_df_protist_stat_total_col <-left_join(division_df_protist_stat_total, classes_division, by=c("Division"= "division")) %>%
distinct(division_df_protist_stat_total_col, Division, .keep_all = TRUE)

color_division <- as.character(division_df_protist_stat_total_col$color_hex_division)
names(color_division) <- as.character(division_df_protist_stat_total_col$division)
division_df_protist_stat_total_col$Division = with(division_df_protist_stat_total_col, reorder(Division, -mean))


CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_SUM_DIVISION.pdf", width = 4, height = 4)
ggplot(division_df_protist_stat_total_col, aes(x=Division, y=mean)) +
    geom_bar(stat="identity", width = 0.9) +
   # scale_fill_manual(values = color_division) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - SUM")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60)
dev.off()


```


## Summary mean and sd relative abundance of classes in different water masses
### WATER MASSES  #####
```{r, fig.height=6, fig.width=6}

## PRotist DIVISION
division_df_protist_stat_watermass <-  df_protist_division.rel %>%
    filter(light_layer == "Euphotic") %>% ### surface values
    group_by(Water_mass.TS1, Division) %>% 
    add_tally() %>%
    summarise(median = median(n_reads, na.rm = FALSE), mean=mean(n_reads, na.rm = FALSE), sd=sd(n_reads, na.rm = FALSE), n=mean(n, na.rm = FALSE)) # we want to account for zero values as well

 division_df_protist_stat_watermass <- division_df_protist_stat_watermass %>%
  arrange(Water_mass.TS1, -mean) 

write.xlsx(division_df_protist_stat, file = "Tables/division_df_protist_stat_WATER_August.xlsx")

division_df_protist_stat_watermass_col <-left_join(division_df_protist_stat_watermass, classes_division, by=c("Division"= "division")) %>%
distinct(division_df_protist_stat_watermass_col, Division, .keep_all = TRUE)

color_division <- as.character(division_df_protist_stat_watermass_col$color_hex_division)
names(color_division) <- as.character(division_df_protist_stat_watermass_col$division)
division_df_protist_stat_watermass_col$Division = with(division_df_protist_stat_watermass_col, reorder(Division, -mean))

CairoPDF(file = "R.Figures/Bar/Protist_synd_euphotic_top20_SUM_DIVISION_WATERMASS2.pdf", width = 4, height = 9)
ggplot(division_df_protist_stat_watermass_col, aes(x=Division, y=mean)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_division) +
    xlab("") +
    theme_bw() +
    #theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - SUM")) + 
    geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=6)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("% Relative abundance") +
    ylim(-10, 60) +
    facet_wrap(~Water_mass.TS1, nrow = 3)

dev.off()


```

## Plotting BOX-PLOTS of groups relative abundance in different areas and environmental ranks 
```{r, fig.height=6, fig.width=6}


###PLOTTING - BOX PLOT of relative abundances in different areas, and other RANKED ENVIRONMENTAL VARIABLES

  ### WATER MASSES ##

CairoPDF(file = "R.Figures/Class/RelativeAbund_EUPHOTIC_Water_masses_PROTIST-SYND-ALL.pdf", width = 12, height = 8)
  df_protist_synd_class.rel %>%
    #filter(NominalDepth == 1) %>%
    filter(light_layer == "Euphotic") %>%
    #filter(dens_layer == "surface") %>% 
    filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%

    ggplot(aes(x = Water_mass.TS1 , y = n_reads,  fill(Water_mass.TS2), color = Water_mass.TS1)) +
        #geom_point(size = 1) +
       geom_boxplot(outlier.shape = NA) + 
        #geom_boxplot() +
      geom_jitter(size = 1.5, width = 0.2, alpha = 0.2) +
        #scale_x_continuous(limits = c(0, 10)) +
        #ylim(-100, 0) +
        #scale_color_manual(values = water_colors) +
       # geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
        theme_bw() + ggtitle("Class - EUPHOTIC zone PROTIST-SYND % abundance in Water.masses") +
        xlab("Water_mass.TS1")+ ylab("standarized abundance (normalized median sequencing depth)") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 12)) +
        theme(axis.text.y = element_text(size = 12)) +
        facet_wrap(~Class, scales = "free_y")
  dev.off()
  
  
  ### AREAS ##
## reorder areas ##
  
CairoPDF(file = "R.Figures/Class/RelativeAbund_EUPHOTIC_Areas_PROTIST_SYND-ALL.pdf", width = 12, height = 8)
  df_protist_synd_class.rel %>%
    mutate(area.name = fct_reorder(Area2, Latitude.CTD)) %>% ## creating a new variable to reorder factors
    #filter(NominalDepth == 1) %>%
    filter(light_layer == "Euphotic") %>%
    #filter(dens_layer == "surface") %>% 
    filter(Class %in% c("Mamiellophyceae", "Dinophyceae", "Chloropicophyceae","Prymnesiophyceae", "Bacillariophyta", "Pelagophyceae", "Prasino-Clade-V", "Cryptophyceae", "Dictyochophyceae", "Choanoflagellatea", "MAST", "Polycystinea", "Acantharea", "RAD-B", "RAD-A", "Spirotrichea" )) %>%

    ggplot(aes(x = area.name , y = n_reads,  fill(Water_mass.TS1), color = Water_mass.TS1)) +
        #geom_point(size = 1) +
       geom_boxplot(outlier.shape = NA) + 
        #geom_boxplot() +
      geom_jitter(size = 1.5, width = 0.2, alpha = 0.2) +
        #scale_x_continuous(limits = c(0, 10)) +
        #ylim(-100, 0) +
        #scale_color_manual(values = water_colors) +
       # geom_errorbar(aes(ymin = mean-sd, ymax = mean+sd), size = 0.1, width = 0.1) +
        theme_bw() + ggtitle("Class - EUPHOTIC % Abundance AREA") +
        xlab("Area")+ ylab("Relative Abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 12)) +
        theme(axis.text.y = element_text(size = 12)) +
        facet_wrap(~Class, scales = "free_y")
  dev.off()

```

## Plotting distribution of groups  % RELATIVE abundances and environmental factors

## Using Geom_smooth - method = loess model y - x /// Need to explore how to use GAM with multiple environmental parameters

### TEMPERATURE 
```{r, fig.height=10, fig.width=10}

## sample water concentration

  CairoPDF(file = "R.Figures/Class/Environmental/Rel_Abundance/REL_Class_vs_Temp_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  df_protist_synd_class.rel %>%
  #filter(NominalDepth == 1) %>%
  #filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  filter(Water_mass.TS1 != "ANT" )   %>%
  
    ggplot(aes(x = Temperature.CTD , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5, size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER -  % abundance vs TEMPERATURE") +
        xlab("Temperature")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        facet_wrap(~Class, scales = "free") 
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales =  "free_y")

  dev.off()
  
  
```

## NITRATE ##
```{r, fig.height=10, fig.width=6}

## sample water concentration
CairoPDF(file = "R.Figures/Class/Environmental/Rel_Abundance/REL_Class_vs_NO3_MIXEDLAYER-ALL.pdf", width = 14, height =8) 
   df_protist_synd_class.rel %>%
  #filter(NominalDepth == 1) %>%
  #filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  filter(Water_mass.TS1 != "ANT" )   %>%
  
    ggplot(aes(x = NO3 , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5, size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        xlab("Nitrate concentration (umol/L)")+ ylab("normalized abundance") +
        theme_bw() + ggtitle("MIXEDLAYER -  % abundance vs NO3") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        theme(strip.text = element_text(size = 12)) +
        facet_wrap(~Class, scales = "free") 
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales =  "free_y")

  dev.off()
  
  
  ## water column median waterconcentration

  CairoPDF(file = "R.Figures/Class/Environmental/Rel_Abundance/REL_Class_vs_medianNO3_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
   df_protist_synd_class.rel %>%
  #filter(NominalDepth == 1) %>%
  #filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  filter(Water_mass.TS1 != "ANT" )   %>%
  
    ggplot(aes(x = median_NO3 , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - % abundance vs median NO3") +
        xlab("Nitrate concentration (umol/L)")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        theme(strip.text = element_text(size = 12))  +
        facet_wrap(~Class, scales = "free") 
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales =  "free_y")
  dev.off()
  
```
## CHLA ##
```{r, fig.height=10, fig.width=6}

  CairoPDF(file = "R.Figures/Class/Environmental/Rel_Abundance/REL_Class_vs_Chla_MIXEDLAYER-ALL.pdf", width = 14, height =8) 
  df_protist_synd_class.rel %>%
  #filter(NominalDepth == 1) %>%
  #filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  filter(Water_mass.TS1 != "ANT" )   %>%
  
    ggplot(aes(x = ChlaTot , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - % abundance vs Chla") +
        xlab("Chla concentration (ug/L)")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
       facet_wrap(~Class, scales = "free") 
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales =  "free_y")
  dev.off()
  
```
## SF - CHLA normalized abundance  vs  chla percentage ##

```{r, fig.height=10, fig.width=6}

## PICO ##
  CairoPDF(file = "R.Figures/Class/Environmental/Rel_Abundance/REL_Class_vs_SFChla_0.2-2_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  df_protist_synd_class.rel %>%
  #filter(NominalDepth == 1) %>%
  #filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  filter(Water_mass.TS1 != "ANT" )   %>%
  
    ggplot(aes(x = pico_pct , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs Chla 0.2-2") +
        xlab("% Chla 0.2-2 um ")+ ylab("normalized abundance") +
         theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
       facet_wrap(~Class, scales = "free") 
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales =  "free_y")
  
  dev.off()
  
  
```
```{r, fig.height=10, fig.width=6}

## NANO ##
  CairoPDF(file = "R.Figures/Class/Environmental/Rel_Abundance/REL_Class_vs_SFChla_2-20_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  df_protist_synd_class.rel %>%
  #filter(NominalDepth == 1) %>%
  #filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  filter(Water_mass.TS1 != "ANT" )   %>%
  
    ggplot(aes(x = nano_pct , y = n_reads, group = Class, color = Water_mass.TS1)) +
        geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs Chla 2-20") +
        xlab("% Chla 2-20 um ")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
       facet_wrap(~Class, scales = "free") 
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales =  "free_y")
  dev.off()
  
  
```

```{r, fig.height=10, fig.width=6}

## MICRO ##
  CairoPDF(file = "R.Figures/Class/Environmental/Rel_Abundance/REL_Class_vs_SFChla_20_MIXEDLAYER-ALL.pdf", width = 14, height = 8) 
  df_protist_synd_class.rel %>%
  #filter(NominalDepth == 1) %>%
  #filter(light_layer == "Euphotic") %>%
  filter(dens_layer == "surface") %>% 
  filter(Water_mass.TS1 != "ANT" )   %>%
  
    ggplot(aes(x = micro_pct , y = n_reads, group = Class, color = Water_mass.TS1)) +
         geom_point(shape = 21, stroke = 0.5,size = 2) +
        geom_smooth(aes(col=Water_mass.TS1), method = "loess",  formula = y ~ x, se=TRUE, linetype=2, size = 0.25) +
        theme_bw() + ggtitle("MIXEDLAYER - Class normalized abundance vs Chla >20") +
        xlab("% Chla >20 um ")+ ylab("normalized abundance") +
        theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
        theme(axis.text.x = element_text(hjust = 1, size = 14)) +
        theme(axis.text.y = element_text(size = 14)) +
        theme(strip.text = element_text(size = 12)) +
        facet_wrap(~Class, scales = "free") 
        #facet_grid(cols = vars(Water_mass.TS1), rows = vars(Class), scales =  "free_y")
  dev.off()
  
  
```

### SPECIES LEVEL ANALYSIS
# Select most abundant species using PHYLOSEQ - NEED TO BE REFINED

```{r pressure, echo=FALSE}

## Select only 30 Most abundant species based on sequencing depth standarized abundance
#PHYTO - STW
ps_phyto_stw <- ps_phyto_std %>%
subset_samples(Water_mass.TS1 == "STW") %>%
subset_samples(light_layer == "Euphotic")

# Most "present" species - Median of read abundance across samples 
phyto_species.median = tapply(taxa_sums(ps_phyto_stw), tax_table(ps_phyto_stw)[, "Species"], median, na.rm=TRUE)
  top15species_stw_median = names(sort(phyto_species.median, TRUE))[1:20]
  Top15_phyto_stw_median = prune_taxa((tax_table(ps_phyto_stw)[, "Species"] %in% top15species_stw_median), ps_phyto_stw)
  Top15_phyto_stw_median
  
  
# Most abundant species - Total sum of reads abundance across samples 
phyto_species.sum = tapply(taxa_sums(ps_phyto_stw), tax_table(ps_phyto_stw)[, "Species"], sum, na.rm=TRUE)
  top15species_stw_sum = names(sort(phyto_species.sum, TRUE))[1:20]
  Top15_phyto_stw_sum = prune_taxa((tax_table(ps_phyto_stw)[, "Species"] %in% top15species_stw_sum), ps_phyto_stw)
  Top15_phyto_stw_sum
  
Mostabundant = data.frame(nreads = sort(taxa_sums(Top15_phyto_stw_sum), TRUE), sorted = 1:ntaxa(Top15_phyto_stw_sum), Species = tax_table(Top15_phyto_stw_sum)[, "Species" ])

taxo_df <- as.data.frame(Top15_phyto_stw_sum@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

Mostabundant_tax <- left_join(Mostabundant, taxo_df)

    metadata_df <- data.frame(sample_data(ps)) %>%
      rownames_to_column(var = "file_code")

    otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))
  
p = plot_bar(Top15_phyto_stw_sum, "Genus", fill="Class", facet_grid= ~Area)
p + geom_bar(aes(color=Class, fill=Class), stat="identity")

```

# Dataframe and wide to long transformation of phyloseq object - Most abundant PROTIST - SYND species
```{r pressure, echo=FALSE}

    otu_df <- as.data.frame(ps_protist_synd_std@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      #filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

  # See https://github.com/joey711/phyloseq/issues/983
    taxo_df <- as.data.frame(ps_protist_synd_std@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

    otu_df <- left_join(taxo_df, otu_df)

    metadata_df <- data.frame(sample_data(ps_protist_std)) %>%
      rownames_to_column(var = "file_code")

    otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))
    
    otu_df_protist_synd <-left_join(otu_df, classes_division, by=c("Class"= "class"))
    
    write.xlsx(otu_df_protist_synd, file = "Tables/otu_df_protist_synd.xlsx")

```

# Dataframe and wide to long transformation of phyloseq object - For selection of most abundant PHYTO species
```{r pressure, echo=FALSE}

    otu_df <- as.data.frame(ps_phyto_std@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      #filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

  # See https://github.com/joey711/phyloseq/issues/983
    taxo_df <- as.data.frame(ps_phyto_std@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

    otu_df <- left_join(taxo_df, otu_df)

    metadata_df <- data.frame(sample_data(ps_phyto_std)) %>%
      rownames_to_column(var = "file_code")

    otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))
    
    otu_df_phyto <-left_join(otu_df, classes_division, by=c("Class" = "class")) 

     write.xlsx(otu_df_phyto, file = "Tables/otu_df_phyto.xlsx")

```

# Select most abundant OTUs per WATER MASSES using dplyr - EUPHOTIC

```{r, fig.height=8, fig.width=8}
OTU_abund <- otu_df_protist_synd %>%
  filter(light_layer == "Euphotic") %>%
  group_by(OTUNumber, Water_mass.TS1) %>%
  mutate(mean_OTU = mean(n_reads, na.rm=TRUE), median_OTU = median(n_reads, na.rm=TRUE), Total_OTU = sum(n_reads, na.rm = TRUE)) %>%
  select(OTUNumber:Water_mass.TS1, Sample_Type, light_layer, supergroup:Total_OTU) %>%
  arrange(Water_mass.TS1, -Total_OTU) %>%
  unite(OTU_water, OTUNumber, Water_mass.TS1, remove = FALSE) %>% # to allow selecting for distinct OTUs
  unite(OTU_genus, OTUNumber, Genus, remove = FALSE) %>%
  unite(OTU_species, OTUNumber, Species, remove = FALSE)


## Select most abundant OTUs for plotting SUM TOTAL read abundance values
Top_otu_sum <- OTU_abund %>%
  distinct(OTU_water, .keep_all = TRUE) %>%
  arrange(Water_mass.TS1, -Total_OTU) %>%
   group_by(Water_mass.TS1) %>% 
   top_n(20,Total_OTU)

## Select most abundant OTUs for plotting MEAN read abundance values
Top_otu_mean <- OTU_abund %>%
  distinct(OTU_water, .keep_all = TRUE) %>%
     arrange(Water_mass.TS1, -mean_OTU) %>%
     group_by(Water_mass.TS1) %>% 
     top_n(20, mean_OTU)

## Select most abundant OTUs for plotting MEDIAN read abundance values
Top_otu_median <- OTU_abund %>%
  distinct(OTU_water, .keep_all = TRUE) %>%
     arrange(Water_mass.TS1, -median_OTU) %>%
     group_by(Water_mass.TS1) %>% 
    top_n(20,median_OTU)

    #select(-median_OTU, median_OTU) %>% #relocate(median_OTU, .after = last_col()) %>%


write.xlsx(Top_otu_sum, file = "Tables/Protist_synd_watermass_EUPHOTIC_top20_SUM.xlsx")
write.xlsx(Top_otu_median, file = "Tables/Protist_synd_watermass_EUPHOTIC_top20_MEDIAN.xlsx")
write.xlsx(Top_otu_mean, file = "Tables/Protist_synd_watermass_EUPHOTIC_top20_MEAN.xlsx")



```

# Select most abundant OTUs per AREA using dplyr

```{r, fig.height=8, fig.width=8}
OTU_abund_area <- otu_df_protist %>%
  filter(light_layer == "Euphotic") %>%
  group_by(OTUNumber, Area2) %>%
  mutate(median_OTU_area = median(n_reads, na.rm=TRUE), Total_OTU_area = sum(n_reads, na.rm = TRUE)) %>%
  select(OTUNumber:Water_mass.TS1, Sample_Type, light_layer, supergroup:Total_OTU_area) %>%
  arrange(Area2, -Total_OTU_area) %>%
  unite(OTU_water, OTUNumber, Area2, remove = FALSE) %>% # to allow selecting for distinct OTUs
  unite(OTU_genus, OTUNumber, Genus, remove = FALSE) %>%
  unite(OTU_species, OTUNumber, Species, remove = FALSE)


## Select most abundant OTUs for plotting SUM TOTAL abundance values
Top_otu_area_sum <- OTU_abund_area %>%
  distinct(OTU_water, .keep_all = TRUE) %>%
   group_by(Area2) %>% 
   top_n(20,Total_OTU_area)

## Select most abundant OTUs for plotting MEDIAN abundance values
Top_otu_area_median <- OTU_abund_area %>%
  distinct(OTU_water, .keep_all = TRUE) %>%
  arrange(Area2, -median_OTU_area) %>%
  group_by(Area2) %>% 
  top_n(20,median_OTU_area)

write.xlsx(Top_otu_area_sum, file = "Tables/Protist_Area_Euphotic_top20_TOTAL.xlsx")
write.xlsx(Top_otu_area_median, file = "Tables/Protist_Area_Euphotic_top20_MEDIAN.xlsx")

```



# Plotting  most abundant OTUs per WATER MASSES using dplyr

```{r, fig.height=8, fig.width=8}


# color coding # Water mass - SUM
color_class1 <- as.character(Top_otu_sum$color_hex_class)
names(color_class1) <- as.character(Top_otu_sum$Class)
Top_otu_sum$OTU_species = with(Top_otu_sum, reorder(OTU_species, Total_OTU))

# color coding # Water mass - MEDIAN
color_class2 <- as.character(Top_otu_median$color_hex_class)
names(color_class2) <- as.character(Top_otu_median$Class)
Top_otu_median$OTU_species = with(Top_otu_median, reorder(OTU_species, median_OTU))

# color coding # Water mass - MEAN
color_class3 <- as.character(Top_otu_mean$color_hex_class)
names(color_class3) <- as.character(Top_otu_mean$Class)
Top_otu_mean$OTU_species = with(Top_otu_mean, reorder(OTU_species, mean_OTU))

# color coding # AREA - SUM
color_class4 <- as.character(Top_otu_area_sum$color_hex_class)
names(color_class4) <- as.character(Top_otu_area_sum$Class)
Top_otu_area_sum$OTU_species = with(Top_otu_area_sum, reorder(OTU_species, Total_OTU_area))

# color coding # AREA - MEDIAN
color_class5 <- as.character(Top_otu_area_median$color_hex_class)
names(color_class5) <- as.character(Top_otu_area_median$Class)
Top_otu_area_median$OTU_species = with(Top_otu_area_median, reorder(OTU_species, median_OTU_area))


## Plot without reordering using facet_wrap - Problems reordering withing each water mass

#CairoPDF(file = "R.Figures/Species/Protist_Aphotic_top20b.pdf", width = 8, height =12)
#pdf(file="R.Figures/Species/Protist_euphotic_top20.pdf")
ggplot(Top_otu_sum, aes(x=OTU_species, y=Total_OTU, fill=Class)) +
    geom_bar(stat="identity", width = 0.9) +
    scale_fill_manual(values = color_class1) +
    xlab("") +
    theme_bw() +
    theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist-synd - Euphotic - SUM")) + 
    theme(axis.text.x=element_text(angle = 45, hjust = 1, size=6)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=4)) +
    theme(panel.grid.major = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey"), 
    panel.grid.minor = element_line(size = 0.05, linetype = 'solid',
                                colour = "grey")) +
    theme(axis.ticks = element_line(colour = "black", size = 0.05)) +
    xlab("") + 
    ylab("Cumulative OTU - Sequencing depth standarized abundance") +
    coord_flip() +
    facet_wrap(~Water_mass.TS1, scales="free")
 #dev.off()
 
  ggsave("Protist_synd_euphotic_top20_SUM_OTUs.pdf", path = "R.Figures/Species/water_mass", dpi = 300, height = 4, width = 8)
  
## Reordering and plotting using forcat package - One water mass at at time

 Top_otu_order <- Top_otu %>%
   filter(Water_mass.TS1 == "STW") %>%
   mutate(name = fct_reorder(OTU_genus, Total_OTU)) 
   
 CairoPDF(file = "R.Figures/Species/Protist_euphotic_top20_STW.pdf", width = 12, height = 16) 
 ggplot(Top_otu_order, aes(x=name, y=Total_OTU, fill=Class)) +
    geom_bar(stat = "identity", width = 0.95) +
    coord_flip() +
    #scale_fill_viridis_d(option = "plasma") +
    scale_fill_manual(values = color_class) +
    xlab("") +
    theme_bw() +
    theme(aspect.ratio = 2/1) +
    ggtitle(paste (" Protist - Euphotic - STW")) + 
    theme(axis.text.x=element_text(angle = 0, hjust = 1, size=12)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=12)) +
    xlab("") + 
    ylab("Cumulative OTU - Sequencing depth standarized abundance")
    #facet_wrap(~Water_mass.TS1, scales="free")
 dev.off()


### HAve problems to install Slice function from hadleys github

# Top_otu <- OTU_abund %>%
#   group_by(Water_mass.TS1) %>%
#   slice_max(Total_OTU, n = 10)

```

# Plotting  most abundant OTUs per REGIONS using dplyr

```{r, fig.height=8, fig.width=8}

# color coding
color_class <- as.character(Top_otu$color_hex_class)
names(color_class) <- as.character(Top_otu$Class)
Top_otu$OTU_genus = with(Top_otu, reorder(OTU_genus, Total_OTU_area))

## Plot without reordering using facet_wrap - Problems reordering withing each water mass

CairoPDF(file = "R.Figures/Species/Protist_aphotic_regions_top20.pdf", width = 8, height =12) 
#pdf(file="R.Figures/Species/Protist_euphotic_top20.pdf")
ggplot(Top_otu, aes(x=OTU_genus, y=Total_OTU_area, fill=Class)) +
    geom_bar(stat="identity", width = 0.95) +
    scale_fill_manual(values = color_class) +
    xlab("") +
    theme_bw() +
    theme(aspect.ratio = 2/1) +
    ggtitle(paste ("Protist - Euphotic - Sum")) + 
    theme(axis.text.x=element_text(angle = 0, hjust = 1, size=12)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=12)) +
    xlab("") + 
    ylab("Cumulative OTU - Sequencing depth standarized abundance") +
    coord_flip() +
    facet_wrap(~Area2, scales="free")
 dev.off()
 
   # ggsave("Protist_euphotic_top20.pdf", path = "R.Figures/Species", dpi = 300,  width = 8, height = 16, scale=1.2)
  
## Reordering and plotting using forcat package - One water mass at at time

 Top_otu_order <- Top_otu %>%
   filter(Area2 == "Bay of Plenty") %>%
   mutate(name = fct_reorder(OTU_genus, Total_OTU_area)) 
   
 CairoPDF(file = "R.Figures/Species/Regions/Phyto_euphotic_top20_BayofPlenty.pdf", width = 12, height = 16) 
 ggplot(Top_otu_order, aes(x=name, y=Total_OTU_area, fill=Class)) +
    geom_bar(stat = "identity", width = 0.95) +
    coord_flip() +
    #scale_fill_viridis_d(option = "plasma") +
    scale_fill_manual(values = color_class) +
    xlab("") +
    theme_bw() +
    theme(aspect.ratio = 2/1) +
    ggtitle(paste (" Phyto - Euphotic - Bay of Plenty")) + 
    theme(axis.text.x=element_text(angle = 0, hjust = 1, size=12)) +
    theme(axis.text.y=element_text(angle = 0, hjust = 1, size=12)) +
    xlab("") + 
    ylab("Cumulative OTU - Sequencing depth standarized abundance")
    #facet_wrap(~Water_mass.TS1, scales="free")
 dev.off()


### HAve problems to install Slice function from hadleys github

# Top_otu <- OTU_abund %>%
#   group_by(Water_mass.TS1) %>%
#   slice_max(Total_OTU, n = 10)

```


