---
title: "TREEMAPS - PHYLOSEQ - CATALYST"
author: "Andres"
date: "14/05/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Libraries

```{r cars}
  library(stringr)
   library(ggplot2)
   library(dplyr)
   library(tidyr)
   library(tibble)
   library(readr)
   library(maps)
   library(glue)
   library("cowplot")
   library("treemapify")
library("Cairo")
library("cairoDevice")
library("readxl")
library("openxlsx")
library("gplots")
library(forcats)
library(mgcv)
   library(phyloseq)
library(data.table)
library(viridis)

   # library(pr2database)
   # data("pr2")
```

#Load the phyloseq files

```{r pressure, echo=FALSE}
  
  ps <- list()
  ps[["all"]] <- readRDS("intomob123_phyloseq_15March2021_RANKED.rds") 
  
```
  

# Split samples into subsets - Sample types

```{r pressure, echo=FALSE}
  #Very important, must remove taxa that are not present in one sample set
  #creating sample_types
  sample_types = c("Seawater", "Trap", "Sediment")
  
  ps[["Seawater"]] <-  ps[["all"]] %>%  
    subset_samples(Sample_Type == "Seawater")
  ps[["Trap"]] <-  ps[["all"]] %>%  
    subset_samples(Sample_Type == "Trap")
  ps[["Sediment"]] <-  ps[["all"]] %>%  
    subset_samples(Sample_Type == "Sediment")
  
  # sample_sums( ps[["Seawater"]])
  # median(sample_sums( ps[["Seawater"]]))

```


##Subsampled object - Only seawater samples - remove data missing metadata
```{r}

ps[["EEZ"]] <- ps[["Seawater"]] %>%
              subset_samples(Sampling_Type != "Blank" ) %>%
              subset_samples(Water_mass.TS1 != "ANT" )  ### Remove antarctic samples but keep subantarctic samples from antarctic TAN1802 trip
              #subset_samples(Project != "SOAP") %>% 
              #subset_samples(Project != "Bay of Plenty") ## remove samples from these voyages until I get the metadata (CTD, Nutrients and Chla) completed - if I{ dont receive those data I am inclined to leave these samples out of the analysis completely 
              
median(sample_sums(ps[["EEZ"]]))

```

# Filtration based on Taxonomy - for Protist, Protist-synd and Phyto
## I am doing this filtration later in the workflow - but not sure whether it matters doing it before or after normalizing the number of reads to the median sequencing depth or not.

```{r pressure, echo=FALSE}
  # Remove Metazoa
  # Remove Fungi

  ## Protist
  ps[["EEZ"]] <- ps[["EEZ"]] %>% 
        subset_taxa(!(Division %in% c("Metazoa", "Fungi"))) 

  ## Protist - syndiniales and sarcomonadae
  ps[["EEZ-synd"]] <- ps[["EEZ"]] %>% 
        subset_taxa(!(Division %in% c("Metazoa", "Fungi"))) %>%
      subset_taxa(!(Class %in% c("Syndiniales", "Sarcomonadea")))
  
  ## Phyto
  ps[["EEZ-phyto"]] <- ps[["EEZ"]] %>%
        subset_taxa(!(Division %in% c("Metazoa", "Fungi"))) %>%
        subset_taxa(Division %in% c("Chlorophyta", "Dinoflagellata", "Cryptophyta", "Haptophyta", "Ochrophyta", "Cercozoa")) %>%
        subset_taxa(!(Class %in% c("Syndiniales", "Sarcomonadea")))


# get_taxa_unique(ps[["EEZ-synd"]], "Class")


```






# Normalize and transform to long form
# Define function - phyloseq_transform_to_long
```{r pressure, echo=FALSE}
  phyloseq_transform_to_long <- function(ps) {
    otu_df <- as.data.frame(ps@otu_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber") %>%
      pivot_longer(cols = -OTUNumber,
                   names_to = "file_code",
                   values_to = "n_reads",
                   values_drop_na = TRUE) %>%
      filter(n_reads != 0) %>%
      filter(!is.na(n_reads))

  # See https://github.com/joey711/phyloseq/issues/983
    taxo_df <- as.data.frame(ps@tax_table@.Data, stringsAsFactors = FALSE) %>%
      rownames_to_column(var = "OTUNumber")

    otu_df <- left_join(taxo_df, otu_df)

    metadata_df <- data.frame(sample_data(ps)) %>%
      rownames_to_column(var = "file_code")

    otu_df <- left_join(otu_df, metadata_df, by = c("file_code"))
    
    #otu_df <-merge(otu_df, classes_division, by=c("Division"))


    return(otu_df)

  }

```
# Normalize 
# Define function - phyloseq_normalize_median
```{r pressure, echo=FALSE}
 phyloseq_normalize_median <- function (ps) {
  ps_median = median(sample_sums(ps))
  normalize_median = function(x, t=ps_median) (if(sum(x) > 0){ round(t * (x / sum(x)))} else {x})
  ps = transform_sample_counts(ps, normalize_median)
  cat(str_c("\n========== \n") )
  print(ps)
  cat(sprintf("\n==========\nThe median number of reads used for normalization is  %.0f", ps_median))
  return(ps)
}

```

## ALL PROTISTS

# Split SEAWATER samples into subsets - Water_mass.TS1


```{r pressure, echo=FALSE}
  #Very important, must remove taxa that are not present in one sample set
  #creating sample_types
  Water_mass.TS1 = c("STW", "STF", "SAW")

  ps[["STW"]] <-  ps[["EEZ"]] %>%  
    subset_samples(Water_mass.TS1 == "STW")
  ps[["STF"]] <-  ps[["EEZ"]] %>%  
    subset_samples(Water_mass.TS1 == "STF")
  ps[["SAW"]] <-  ps[["EEZ"]] %>%  
    subset_samples(Water_mass.TS1 == "SAW")
  
  
  
 # Median number of reads in samples across different water masses
  for (one_sample_type in Water_mass.TS1) {
    
    
      nnn <- median(sample_sums(ps[[one_sample_type]])  )
      print(glue("Phyloseq - {one_sample_type}"))
      print(nnn)
      cat("============================\n")
      
  }
  
  # Remove ASVs that are not present in the dataset
  for (one_sample_type in Water_mass.TS1) {

      ps[[one_sample_type]] <- ps[[one_sample_type]] %>% 
        filter_taxa(function(x) sum(x) > 0 , TRUE) 
      
     
      print(glue("Phyloseq - {one_sample_type}"))
      print(ps[[one_sample_type]] )
      cat("============================\n")
      
  }
```

# Do normalizations and transformation - WATER MASS

```{r pressure, echo=FALSE}
  long_water_mass.TS1 <- list()

  for (one_sample_type in Water_mass.TS1) {
    
      cat(one_sample_type)
      cat("\n============================")  
      
      ps[[one_sample_type]] <- phyloseq_normalize_median(ps[[one_sample_type]])
      
      long_water_mass.TS1[[one_sample_type]] <- phyloseq_transform_to_long(ps[[one_sample_type]])
      
      cat("\n") 
      
  }
```

# Creating lists for some specific cases - WATER MASSES - EUPHOTIC


```{r pressure, echo=FALSE}
 water_mass_euphotic= c("EEZ_euphotic","STW_euphotic", "STF_euphotic", "SAW_euphotic")

ps[["EEZ_euphotic"]] <-  ps[["EEZ"]] %>%  subset_samples(light_layer== "Euphotic")

ps[["STW_euphotic"]] <-  ps[["STW"]] %>%  subset_samples(light_layer== "Euphotic") 

ps[["STF_euphotic"]] <-  ps[["STF"]] %>%  subset_samples(light_layer== "Euphotic") 

ps[["SAW_euphotic"]] <-  ps[["SAW"]] %>%  subset_samples(light_layer== "Euphotic") 


for (one_sample_type in water_mass_euphotic) {

      
      print(glue("Phyloseq - {one_sample_type}"))
      print(ps[[one_sample_type]] )
      cat("============================\n")
      
}

```


# Do TREEMAPS
# Define function

```{r pressure, echo=FALSE}
  phyloseq_long_treemap <- function(df, group1, group2, title) {

     df <- df %>%
       group_by({{group1}}, {{group2}}) %>%
       summarise(n_reads=sum(n_reads, na.rm = TRUE))
     g_treemap <- ggplot(df, aes(area = n_reads,
                                 fill = {{group2}},
                                 label = {{group2}},
                                 subgroup = {{group1}})) +
        ggtitle(title) +
        treemapify::geom_treemap() +
        treemapify::geom_treemap_subgroup_border() +
        treemapify::geom_treemap_text(colour = "black", place = "topleft", reflow = T,
                                      padding.x =  grid::unit(1, "mm"),
                                      padding.y = grid::unit(1, "mm") ) +
        treemapify::geom_treemap_subgroup_text(place = "centre", grow = T, alpha = 0.25, colour =
                             "white", fontface = "italic", min.size = 0) +
        scale_fill_viridis_d() +
        theme(legend.position="none", plot.title = element_text(size = 12, face = "bold"))
     print(g_treemap)
     return(list(df, g_treemap)) #for using with cowplot the plots need to save as objects. Instead of return(df), we have to return the df and                                   the plots = g_treemap 
 }
```



# Plot Treemap - WATER MASSES - EUPHOTIC


```{r pressure, echo=FALSE}
long_grouped_class <- list()
long_grouped_genus <- list()

for (one_sample_type in water_mass_euphotic) {
    
      long_water_mass.TS1[[one_sample_type]] <- phyloseq_transform_to_long(ps[[one_sample_type]])
  }

  
for (one_sample_type in water_mass_euphotic) {
    
      long_grouped_class[[one_sample_type]] <- phyloseq_long_treemap(long_water_mass.TS1[[one_sample_type]], 
                                                                     Division, Class, 
                                                                     str_c(one_sample_type, " - Class" ))
      
       long_grouped_genus[[one_sample_type]] <- phyloseq_long_treemap(long_water_mass.TS1[[one_sample_type]],
                                                                    Class, Genus, 
                                                                     str_c(one_sample_type, " - Genus" ))
 
  }



```


# Class x genus WATER MASSES - EUPHOTIC
```{r pressure, echo=FALSE}
CairoPDF(file = "R.Figures/Treemaps/Treemap_watermass_EUPHOTIC_PROTIST_CLASS_GENUS", width = 10, height = 16) 
plot_grid(long_grouped_class[["EEZ_euphotic"]][[2]], long_grouped_genus[["EEZ_euphotic"]][[2]],long_grouped_class[["STW_euphotic"]][[2]], long_grouped_genus[["STW_euphotic"]][[2]], long_grouped_class[["STF_euphotic"]][[2]], long_grouped_genus[["STF_euphotic"]][[2]], long_grouped_class[["SAW_euphotic"]][[2]], long_grouped_genus[["SAW_euphotic"]][[2]], nrow = 4, ncol = 2, rel_heights= c(1,1) )
dev.off()


```
#######################
## PROTIST - SYND 
#######################
# Creating lists for some specific cases - WATER MASSES - EUPHOTIC
```{r pressure, echo=FALSE}
  #Very important, must remove taxa that are not present in one sample set
  #creating sample_types
  Water_mass.TS1_synd = c("STW-synd", "STF-synd", "SAW-synd")

  ps[["STW-synd"]] <-  ps[["EEZ-synd"]] %>%  
    subset_samples(Water_mass.TS1 == "STW")
  ps[["STF-synd"]] <-  ps[["EEZ-synd"]] %>%  
    subset_samples(Water_mass.TS1 == "STF")
  ps[["SAW-synd"]] <-  ps[["EEZ-synd"]] %>%  
    subset_samples(Water_mass.TS1 == "SAW")
  
  
  
 # Median number of reads in samples across different water masses
  for (one_sample_type in Water_mass.TS1_synd) {
    
    
      nnn <- median(sample_sums(ps[[one_sample_type]])  )
      print(glue("Phyloseq - {one_sample_type}"))
      print(nnn)
      cat("============================\n")
      
  }
  
  # Remove ASVs that are not present in the dataset
  for (one_sample_type in Water_mass.TS1_synd) {

      ps[[one_sample_type]] <- ps[[one_sample_type]] %>% 
        filter_taxa(function(x) sum(x) > 0 , TRUE) 
      
     
      print(glue("Phyloseq - {one_sample_type}"))
      print(ps[[one_sample_type]] )
      cat("============================\n")
      
  }
```

# Creating lists for some specific cases - WATER MASSES - EUPHOTIC

```{r pressure, echo=FALSE}
 water_mass_euphotic_synd= c("EEZ_euphotic-synd","STW_euphotic-synd", "STF_euphotic-synd", "SAW_euphotic-synd")

ps[["EEZ_euphotic-synd"]] <-  ps[["EEZ-synd"]] %>%  subset_samples(light_layer== "Euphotic")

ps[["STW_euphotic-synd"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic") 

ps[["STF_euphotic-synd"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") 

ps[["SAW_euphotic-synd"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") 


for (one_sample_type in water_mass_euphotic_synd) {

      
      print(glue("Phyloseq - {one_sample_type}"))
      print(ps[[one_sample_type]] )
      cat("============================\n")
      
}

```



# Plot Treemap - WATER MASSES - EUPHOTIC


```{r pressure, echo=FALSE}
long_grouped_class <- list()
long_grouped_genus <- list()

for (one_sample_type in water_mass_euphotic_synd) {
    
      long_water_mass.TS1[[one_sample_type]] <- phyloseq_transform_to_long(ps[[one_sample_type]])
  }

  
for (one_sample_type in water_mass_euphotic_synd) {
    
      long_grouped_class[[one_sample_type]] <- phyloseq_long_treemap(long_water_mass.TS1[[one_sample_type]], 
                                                                     Division, Class, 
                                                                     str_c(one_sample_type, " - Class" ))
      
       long_grouped_genus[[one_sample_type]] <- phyloseq_long_treemap(long_water_mass.TS1[[one_sample_type]],
                                                                    Class, Genus, 
                                                                     str_c(one_sample_type, " - Genus" ))
 
  }



```


# Class x genus WATER MASSES - EUPHOTIC
```{r pressure, echo=FALSE}
CairoPDF(file = "R.Figures/Treemaps/Treemap_watermass_EUPHOTIC_SYND_CLASS_GENUS", width = 10, height = 16) 
plot_grid(long_grouped_class[["EEZ_euphotic-synd"]][[2]], long_grouped_genus[["EEZ_euphotic-synd"]][[2]],long_grouped_class[["STW_euphotic-synd"]][[2]], long_grouped_genus[["STW_euphotic-synd"]][[2]], long_grouped_class[["STF_euphotic-synd"]][[2]], long_grouped_genus[["STF_euphotic-synd"]][[2]], long_grouped_class[["SAW_euphotic-synd"]][[2]], long_grouped_genus[["SAW_euphotic-synd"]][[2]], nrow = 4, ncol = 2, rel_heights= c(1,1) )
dev.off()

```

# Creating lists for some specific TAXONOMIC GROUPS - WATER MASSES - EUPHOTIC

# Suppl Fig S11

```{r pressure, echo=FALSE}
#get_taxa_unique(ps[["Seawater"]], "Class")

water_mass_group= c("STW_chloro", "STF_chloro", "SAW_chloro", "STW_ochro", "STF_ochro", "SAW_ochro","STW_hapto", "STF_hapto", "SAW_hapto", "STW_crypto", "STF_crypto", "SAW_crypto","STW_dino", "STF_dino", "SAW_dino", "STW_rads", "STF_rads", "SAW_rads", "STW_MAST", "STF_MAST", "SAW_MAST", "STW_ciliates", "STF_ciliates", "SAW_ciliates" , "STW_dicty", "STF_dicty", "SAW_dicty")

ps[["STW_chloro"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Chlorophyta"))) 

ps[["STF_chloro"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Chlorophyta"))) 

ps[["SAW_chloro"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Chlorophyta"))) 

ps[["STW_ochro"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Ochrophyta"))) 

ps[["STF_ochro"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Ochrophyta"))) 

ps[["SAW_ochro"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Ochrophyta"))) 

ps[["STW_hapto"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Haptophyta"))) 

ps[["STF_hapto"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Haptophyta"))) 

ps[["SAW_hapto"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Haptophyta"))) 

ps[["STW_crypto"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Cryptophyta"))) 

ps[["STF_crypto"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Cryptophyta"))) 

ps[["SAW_crypto"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Cryptophyta"))) 

ps[["STW_dino"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Dinoflagellata"))) 

ps[["STF_dino"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Dinoflagellata"))) 

ps[["SAW_dino"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Dinoflagellata"))) 

ps[["STW_rads"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Radiolaria"))) 

ps[["STF_rads"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Radiolaria"))) 

ps[["SAW_rads"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Radiolaria"))) 

ps[["STW_MAST"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Stramenopiles_X"))) 

ps[["STF_MAST"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Stramenopiles_X"))) 

ps[["SAW_MAST"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Stramenopiles_X"))) 

ps[["STW_ciliates"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Division %in% c("Ciliophora"))) 

ps[["STF_ciliates"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Ciliophora"))) 

ps[["SAW_ciliates"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Division %in% c("Ciliophora"))) 

ps[["STW_dicty"]] <-  ps[["STW-synd"]] %>%  subset_samples(light_layer== "Euphotic")  %>% 
                   subset_taxa((Class %in% c("Dictyochophyceae"))) 

ps[["STF_dicty"]] <-  ps[["STF-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Class %in% c("Dictyochophyceae"))) 

ps[["SAW_dicty"]] <-  ps[["SAW-synd"]] %>%  subset_samples(light_layer== "Euphotic") %>% 
                   subset_taxa((Class %in% c("Dictyochophyceae"))) 


for (one_sample_type in water_mass_group) {

      
      print(glue("Phyloseq - {one_sample_type}"))
      print(ps[[one_sample_type]] )
      cat("============================\n")
      
}

```



# Plot Treemap - WATER MASSES - EUPHOTIC - ZOOMING TAXONOMIC CLASSES 


```{r pressure, echo=FALSE}
# long_grouped_class <- list()
long_grouped_genus <- list()

for (one_sample_type in water_mass_group) {
    
      long_water_mass.TS1[[one_sample_type]] <- phyloseq_transform_to_long(ps[[one_sample_type]])
  }

  
for (one_sample_type in water_mass_group) {
    

      
       long_grouped_genus[[one_sample_type]] <- phyloseq_long_treemap(long_water_mass.TS1[[one_sample_type]],
                                                                    Class, Species, 
                                                                     str_c(one_sample_type, " - Genus" ))
 
  }


```

# Genus TAXONOmIC GROUPS - EUPHOTIC - WATER MASSES
```{r pressure, echo=FALSE}

### Genus

CairoPDF(file = "R.Figures/Treemaps/Groups_watermass_large_species", width = 30, height = 50) 
plot_grid(long_grouped_genus[["STW_chloro"]][[2]], long_grouped_genus[["STF_chloro"]][[2]],long_grouped_genus[["SAW_chloro"]][[2]], long_grouped_genus[["STW_ochro"]][[2]], long_grouped_genus[["STF_ochro"]][[2]],long_grouped_genus[["SAW_ochro"]][[2]], long_grouped_genus[["STW_hapto"]][[2]], long_grouped_genus[["STF_hapto"]][[2]],long_grouped_genus[["SAW_hapto"]][[2]], long_grouped_genus[["STW_dino"]][[2]], long_grouped_genus[["STF_dino"]][[2]],long_grouped_genus[["SAW_dino"]][[2]], long_grouped_genus[["STW_crypto"]][[2]], long_grouped_genus[["STF_crypto"]][[2]],long_grouped_genus[["SAW_crypto"]][[2]], long_grouped_genus[["STW_rads"]][[2]], long_grouped_genus[["STF_rads"]][[2]],long_grouped_genus[["SAW_rads"]][[2]], 
long_grouped_genus[["STW_MAST"]][[2]], long_grouped_genus[["STF_MAST"]][[2]],long_grouped_genus[["SAW_MAST"]][[2]],
long_grouped_genus[["STW_ciliates"]][[2]], long_grouped_genus[["STF_ciliates"]][[2]],long_grouped_genus[["SAW_ciliates"]][[2]], 
long_grouped_genus[["STW_dicty"]][[2]], long_grouped_genus[["STF_dicty"]][[2]],long_grouped_genus[["SAW_dicty"]][[2]], nrow = 9, ncol = 3, rel_heights=2 )

dev.off()

```

# NOT USED IN MARCH2021 VERSION OF MS
########################
## PHYTO ## 
#########################

# Creating lists for some specific cases - WATER MASSES - EUPHOTIC
```{r pressure, echo=FALSE}
  #Very important, must remove taxa that are not present in one sample set
  #creating sample_types
  Water_mass.TS1_phyto = c("STW-phyto", "STF-phyto", "SAW-phyto")

  ps[["STW-phyto"]] <-  ps[["EEZ-phyto"]] %>%  
    subset_samples(Water_mass.TS1 == "STW")
  ps[["STF-phyto"]] <-  ps[["EEZ-phyto"]] %>%  
    subset_samples(Water_mass.TS1 == "STF")
  ps[["SAW-phyto"]] <-  ps[["EEZ-phyto"]] %>%  
    subset_samples(Water_mass.TS1 == "SAW")
  
  
  
 # Median number of reads in samples across different water masses
  for (one_sample_type in Water_mass.TS1_phyto) {
    
    
      nnn <- median(sample_sums(ps[[one_sample_type]])  )
      print(glue("Phyloseq - {one_sample_type}"))
      print(nnn)
      cat("============================\n")
      
  }
  
  # Remove ASVs that are not present in the dataset
  for (one_sample_type in Water_mass.TS1_phyto) {

      ps[[one_sample_type]] <- ps[[one_sample_type]] %>% 
        filter_taxa(function(x) sum(x) > 0 , TRUE) 
      
     
      print(glue("Phyloseq - {one_sample_type}"))
      print(ps[[one_sample_type]] )
      cat("============================\n")
      
  }
```



```{r pressure, echo=FALSE}
 water_mass_euphotic_phyto= c("EEZ_euphotic-phyto","STW_euphotic-phyto", "STF_euphotic-phyto", "SAW_euphotic-phyto")

ps[["EEZ_euphotic-phyto"]] <-  ps[["EEZ-phyto"]] %>%  subset_samples(light_layer== "Euphotic")

ps[["STW_euphotic-phyto"]] <-  ps[["STW-phyto"]] %>%  subset_samples(light_layer== "Euphotic") 

ps[["STF_euphotic-phyto"]] <-  ps[["STF-phyto"]] %>%  subset_samples(light_layer== "Euphotic") 

ps[["SAW_euphotic-phyto"]] <-  ps[["SAW-phyto"]] %>%  subset_samples(light_layer== "Euphotic") 


for (one_sample_type in water_mass_euphotic_phyto) {

      
      print(glue("Phyloseq - {one_sample_type}"))
      print(ps[[one_sample_type]] )
      cat("============================\n")
      
}

```




# Plot Treemap - WATER MASSES - EUPHOTIC


```{r pressure, echo=FALSE}
long_grouped_class <- list()
long_grouped_genus <- list()

for (one_sample_type in water_mass_euphotic_phyto) {
    
      long_water_mass.TS1[[one_sample_type]] <- phyloseq_transform_to_long(ps[[one_sample_type]])
  }

  
for (one_sample_type in water_mass_euphotic_phyto) {
    
      long_grouped_class[[one_sample_type]] <- phyloseq_long_treemap(long_water_mass.TS1[[one_sample_type]], 
                                                                     Division, Class, 
                                                                     str_c(one_sample_type, " - Class" ))
      
       long_grouped_genus[[one_sample_type]] <- phyloseq_long_treemap(long_water_mass.TS1[[one_sample_type]],
                                                                    Class, Genus, 
                                                                     str_c(one_sample_type, " - Genus" ))
 
  }



```


# Class x genus WATER MASSES - EUPHOTIC
```{r pressure, echo=FALSE}
CairoPDF(file = "R.Figures/Treemaps/Treemap_watermass_EUPHOTIC_PHYTO_CLASS_GENUS", width = 10, height = 16) 
plot_grid(long_grouped_class[["EEZ_euphotic-phyto"]][[2]], long_grouped_genus[["EEZ_euphotic-phyto"]][[2]],long_grouped_class[["STW_euphotic-phyto"]][[2]], long_grouped_genus[["STW_euphotic-phyto"]][[2]], long_grouped_class[["STF_euphotic-phyto"]][[2]], long_grouped_genus[["STF_euphotic-phyto"]][[2]], long_grouped_class[["SAW_euphotic-phyto"]][[2]], long_grouped_genus[["SAW_euphotic-phyto"]][[2]], nrow = 4, ncol = 2, rel_heights= c(1,1) )
dev.off()

```


